### PR TITLE
refactor: deduplicate structured output testing

### PR DESCRIPTION
**Pull request by teamdandelion**
Now for each provider we do per-mode structured output testing for every
scenario. Where appropriate, we test on multiple models.

Also cleans up some whitespace issues.

### CODE REVIEW

**Comment by teamdandelion** [comment]
* **#1293** <a href="https://app.graphite.dev/github/pr/Mirascope/mirascope/1293?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Mirascope/mirascope/1293?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1288** <a href="https://app.graphite.dev/github/pr/Mirascope/mirascope/1288?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#1300](https://github.com/Mirascope/mirascope/pull/1300) <a href="https://app.graphite.dev/github/pr/Mirascope/mirascope/1300?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1291** <a href="https://app.graphite.dev/github/pr/Mirascope/mirascope/1291?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1287** <a href="https://app.graphite.dev/github/pr/Mirascope/mirascope/1287?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1292** <a href="https://app.graphite.dev/github/pr/Mirascope/mirascope/1292?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1289** <a href="https://app.graphite.dev/github/pr/Mirascope/mirascope/1289?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1290** <a href="https://app.graphite.dev/github/pr/Mirascope/mirascope/1290?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `v2`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python/tests/llm/clients/anthropic/test_anthropic_client.py [hunk 1]
```diff
@@ -33,7 +34,6 @@ def test_call(
 
     kwargs = request.getfixturevalue(scenario_id)
     response = anthropic_client.call(model_id=TEST_MODEL_ID, **kwargs)
-    assert isinstance(response, llm.Response)
```
Comment (teamdandelion) [inline python/tests/llm/clients/anthropic/test_anthropic_client.py:36]: Here (and elsewhere) this was unnecessary, we have type safety for this (and if it went wrong it would be obvious in the snapshots)

File: python/tests/llm/clients/google/test_google_client.py [hunk 2]
```diff
@@ -896,341 +897,805 @@ def test_stream(
             },
         }
     )
+
     assert utils.stream_response_snapshot_dict(stream_response) == expected[scenario_id]
 
 
 @pytest.mark.parametrize("format_mode", FORMATTING_MODES)
 @pytest.mark.parametrize("test_model_id", ["gemini-2.5-flash", "gemini-2.0-flash"])
```
Comment (willbakst) [inline python/tests/llm/clients/google/test_google_client.py:905]: Why does this parameterize on test model id but anthropic doesn't?

File: python/tests/llm/clients/anthropic/test_anthropic_client.py [hunk 3]
```diff
@@ -834,275 +835,493 @@ def test_stream(
...
     assert utils.stream_response_snapshot_dict(stream_response) == expected[scenario_id]
 
 
 @pytest.mark.parametrize("format_mode", FORMATTING_MODES)
+@pytest.mark.parametrize("scenario_id", STRUCTURED_SCENARIO_IDS)
 @pytest.mark.vcr()
-def test_every_structured_mode_basic(
+def test_structured_output_all_scenarios(
     anthropic_client: llm.AnthropicClient,
     format_mode: llm.formatting.FormattingMode,
```
Comment (willbakst) [inline python/tests/llm/clients/anthropic/test_anthropic_client.py:847]: nit: technically this should be named `formatting_mode` to match the class name

File: python/tests/llm/clients/google/test_google_client.py [hunk 4]
```diff
@@ -896,341 +897,805 @@ def test_stream(
...
+@pytest.mark.parametrize("scenario_id", STRUCTURED_SCENARIO_IDS)
 @pytest.mark.vcr()
-def test_every_structured_mode_should_call_tool(
+def test_structured_output_all_scenarios(
     google_client: llm.GoogleClient,
-    format_mode: llm.formatting.FormattingMode,
     test_model_id: llm.clients.google.GoogleModelId,
+    format_mode: llm.formatting.FormattingMode,
```
Comment (willbakst) [inline python/tests/llm/clients/google/test_google_client.py:911]: nit: ditto

File: python/tests/llm/clients/google/test_google_client.py [hunk 5]
```diff
@@ -896,341 +897,805 @@ def test_stream(
...
-    )
+    """Test all structured output scenarios across all formatting modes."""
+    scenario_data = request.getfixturevalue(scenario_id)
     llm.format(scenario_data["format"], mode=format_mode)
 
     try:
         response = google_client.call(model_id=test_model_id, **scenario_data)
-        actual = response.content
+
+        try:
```
Comment (willbakst) [inline python/tests/llm/clients/google/test_google_client.py:922]: nit: should we DRY this into a shared util? seems 100% shared across all tests?

File: python/tests/llm/clients/openai/test_openai_client.py [hunk 6]
```diff
@@ -725,110 +729,93 @@ def test_stream(
...
     )
+
     assert utils.stream_response_snapshot_dict(stream_response) == expected[scenario_id]
 
 
 @pytest.mark.parametrize("format_mode", FORMATTING_MODES)
-@pytest.mark.parametrize(
-    "model_id", ["gpt-4o", "gpt-4"]
-)  # gpt-4o has native strict and json support; gpt-4 does not.
+@pytest.mark.parametrize("test_model_id", ["gpt-4o", "gpt-4"])
```
Comment (willbakst) [inline python/tests/llm/clients/openai/test_openai_client.py:737]: ah, is this because anthropic doesn't have specific models we care to parameterize on for edge cases? I think I would still prefer to match the tests even if anthropic is only parameterized on a single model. Alternatively, if we're specifically seeking error cases, it probably doesn't make sense to duplicate non-error tests since then we would have two of the same tests?

Seems like it might make the most sense to split the error and non-error cases?

File: python/tests/llm/clients/openai/test_openai_client.py [hunk 7]
```diff
@@ -725,110 +729,93 @@ def test_stream(
...
+@pytest.mark.parametrize("scenario_id", STRUCTURED_SCENARIO_IDS)
 @pytest.mark.vcr()
-def test_every_structured_mode_basic(
+def test_structured_output_all_scenarios(
     openai_client: llm.OpenAIClient,
-    model_id: str,
+    test_model_id: str,
     format_mode: llm.formatting.FormattingMode,
```
Comment (willbakst) [inline python/tests/llm/clients/openai/test_openai_client.py:743]: nit: ditto
